index.js

```diff
@@ -57,40 +45,31 @@ class FSMerge {
     let self = this;
     this.fs = new Proxy(nodefs, {
       get(target, propertyName) {
-        if(!WRITEOPERATION.has(propertyName)) {
-          if (READDIR.has(propertyName)) {
-            return function() {
-              let [relativePath] = arguments;
-              if (path.isAbsolute(relativePath) && relativePath.trim() != '/') {
-                return target[propertyName](...arguments);
-              }
-              return self[propertyName](...arguments);
-            }
-          } else if (self[propertyName] && !target[propertyName]) { // when fsMerger.fs.hasOwnProperty is accessed we shouldn't be hijacking it to self.hasOwnProperty
-            return function() {
-              return self[propertyName](...arguments);
-            }
-          } else if (typeof target[propertyName] !== 'function') {
-            return target[propertyName];
-          }
+        if(WHITELISTEDOPERATION.has(propertyName) || self[propertyName]) {
           return function() {
             let [relativePath] = arguments;
-            let { _dirList } = self;
-            let fullPath = relativePath;
             if (!path.isAbsolute(relativePath)) {
```

Comment:
Do we allow path traversals `../../`? If so, should we also disallow those?

Reply:
We can consider them in future iterations if we find issues caused with path traversal.